### PR TITLE
Upstream 2024-06-17

### PR DESCRIPTION
**Pull request by whutchinson98**
Only file with conflict was **libreoffice-core/configure.ac** and it was over the AC_INIT name.

### CODE REVIEW

**Comment by whutchinson98** [comment]
https://www.notion.so/macrocom/Merge-Collabora-Upstream-into-LOK-WASM-ec0f8734e77049ffb1057404d5f1afcc?pvs=4

File: libreoffice-core/desktop/source/lib/init.cxx [hunk 1]
```diff
@@ -3607,9 +3612,42 @@ static int lo_joinThreads(LibreOfficeKit* /* pThis */)
...
     return 1;
 }
 
+static void lo_startThreads(LibreOfficeKit* /* pThis */)
+{
+    auto ucpWebdav = xContext->getServiceManager()->createInstanceWithContext(
+        "com.sun.star.ucb.WebDAVManager", xContext);
+    auto joinable = dynamic_cast<comphelper::LibreOfficeKit::ThreadJoinable *>(ucpWebdav.get());
+    if (joinable)
+        joinable->startThreads();
```
Comment (chase) [inline libreoffice-core/desktop/source/lib/init.cxx:3642]: Just notes for myself for later.

WebDAV stuff should be flagged off for WASM stripping or removed outright. Completely useless to us.

File: libreoffice-core/desktop/source/lib/init.cxx [hunk 2]
```diff
@@ -6607,36 +6650,59 @@ static char* getFonts (const char* pCommand)
...
-    boost::property_tree::ptree aTree;
-    aTree.put("commandName", pCommand);
-    boost::property_tree::ptree aValues;
-    if ( pList )
+    if (!bBloatWithRepeatedSizes)
     {
-        sal_uInt16 nFontCount = pList->GetFontNameCount();
-        for (sal_uInt16 i = 0; i < nFontCount; ++i)
+        tools::JsonWriter aJson;
+        aJson.put("commandName", pCommand);
```
Comment (chase) [inline libreoffice-core/desktop/source/lib/init.cxx:6656]: Font list API changed

File: libreoffice-core/libreofficekit/README.md [hunk 3]
```diff
@@ -81,6 +81,20 @@ once a feature works there, then implement the Android part, with its slower
...
+of the following alternatives, which require no changes to these headers:
+
+- LOK client → core direction: use `postUnoCommand()` to dispatch an UNO command, optionally with
+  parameters.
+- core → LOK client direction:
+  - Use `getCommandValues()` when this is initiated by the LOK client.
+  - Use `LOK_CALLBACK_STATE_CHANGED` with a JSON payload when this is initiated by core.
+
+It's useful to stick to these if possible, only add new C++ API when these are not a good fit.
+
```
Comment (chase) [inline libreoffice-core/libreofficekit/README.md:97]: @whutchinson98 @synoet We're always ahead of the curve lol
Embinding is the future :technologist:

File: libreoffice-core/svx/source/annotation/Annotation.cxx
Comment (chase) [inline libreoffice-core/svx/source/annotation/Annotation.cxx:1]: Maybe affects comment API?

File: libreoffice-core/sw/inc/crsrsh.hxx [hunk 4]
```diff
@@ -285,6 +292,9 @@ private:
     SAL_DLLPRIVATE const SwRangeRedline* GotoRedline_( SwRedlineTable::size_type nArrPos, bool bSelect );
 
     SAL_DLLPRIVATE void sendLOKCursorUpdates();
+
+    DECL_LINK(DoLayoutIdle, Timer*, void); // calls SwViewShell::LayoutIdle
```
Comment (chase) [inline libreoffice-core/sw/inc/crsrsh.hxx:296]: Layout idle changed

File: libreoffice-core/sw/source/core/crsr/crsrsh.cxx [hunk 5]
```diff
@@ -1809,6 +1810,21 @@ void SwCursorShell::VisPortChgd( const SwRect & rRect )
     if( m_bSVCursorVis && bVis ) // show SV cursor again
         m_pVisibleCursor->Show();
 
+    if( comphelper::LibreOfficeKit::isActive() && !rRect.Overlaps( m_aCharRect ))
+    {
+        boost::property_tree::ptree aParams;
```
Comment (chase) [inline libreoffice-core/sw/source/core/crsr/crsrsh.cxx:1815]: API change for cursor update. Also this uses proprerty_tree which is insanely inefficient.

File: libreoffice-core/sw/source/uibase/wrtsh/move.cxx [hunk 6]
```diff
@@ -454,7 +455,14 @@ bool SwWrtShell::PushCursor(SwTwips lOffset, bool bSelect)
     {
         Point aPt( aOldRect.Center() );
 
-        if( !IsCursorVisible() )
+        if( comphelper::LibreOfficeKit::isActive() )
+        {
+            SfxViewShell* pView = GetSfxViewShell();
+            SwRect aLOKVis(pView->getLOKVisibleArea());
```
Comment (chase) [inline libreoffice-core/sw/source/uibase/wrtsh/move.cxx:461]: Probably need to actually set the visible area in our code

**Review by chase — APPROVED** [review]

**Pull request merged**